code/server/data_access/UserMatchDataAccess.js

```diff
@@ -0,0 +1,30 @@
+var UserMatchDB = require('../models/UserMatchDB.js');
+var UserMatchStub = require('../stubs/UserMatchStub.js');
+
+var userMatchDAO;
+
+var init = function(mode) {
+    if (mode == 'stub')
+        userMatchDAO = new UserMatchStub();
+    else if (mode == 'db')
+        userMatchDAO = new UserMatchDB();
+}
```

Comment:
We should probably throw if it doesn't match either. We wouldn't want the program to mysteriously crash later on in the lifecycle because the api DAO ends up being null.